tests/test_dagfactory.py

```diff
@@ -490,6 +490,48 @@ def test_build_dag_with_global_default():
     assert dags.get("example_dag").tasks[0].depends_on_past == True
 
 
+def test_build_dag_with_global_dag_level_defaults():
+    """Test that DAG-level defaults from global defaults.yml are applied to individual DAG configs"""
+    global_defaults = {
+        "default_args": {"owner": "global_owner", "start_date": "2020-01-01",},
+        "schedule": "0 1 * * *",
```

Comment:
```suggestion
        "schedule_interval": "0 1 * * *",
```

Reply:
I've tried this, but it failed when testing with Airflow 3
```
dagfactory.exceptions.DagFactoryException: Failed to generate dag test_dag: 'schedule'
```

I believe it's because this test mock the output of `_global_default_args`, which contains `update_yaml_structure` updating config from Airflow 2 to 3. Hence it should be Airflow 3 keys used in this mock config.